### PR TITLE
Rest api unit tests

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Unit tests for biome rest api

### CODE REVIEW

File: libsplinter/src/biome/user/store/memory.rs [hunk 1]
```diff
@@ -76,6 +89,14 @@ impl UserStore for MemoryUserStore {
             })?;
 
         if inner.remove(id).is_some() {
+            #[cfg(feature = "biome-credentials")]
+            self.credentials_store
+                .remove_credentials(id)
+                .map_err(|err| UserStoreError::QueryError {
+                    context: "Cannot delete credentials store".to_string(),
```
Comment (shannynalayna) [inline libsplinter/src/biome/user/store/memory.rs:96]: I think this error message is slightly off, maybe try rewording to `"Cannot delete {} from credentials store", id`

Comment (RyanLassigBanks) [reply libsplinter/src/biome/user/store/memory.rs]: Changed to `Cannot delete user {} from credentials store`

File: libsplinter/src/biome/rest_api/mod.rs [hunk 2]
```diff
@@ -413,3 +413,793 @@ impl BiomeRestResourceManagerBuilder {
...
+
+    use reqwest::blocking::Client;
+
+    use crate::biome::{
+        MemoryCredentialsStore, MemoryKeyStore, MemoryRefreshTokenStore, MemoryUserStore,
+    };
+    use crate::rest_api::RestApiBuilder;
+
+    static INIT: Once = Once::new();
+    const REST_API_URL: &'static str = "http://127.0.0.1:5000";
```
[the remark this answers is not in the document]
Comment (RyanLassigBanks) [reply libsplinter/src/biome/rest_api/mod.rs]: Updated to bind to port 0

**Comment by vaporos** [comment]
"Add unit tests for biome rest api" commit could have additional summary of what is being tested.

The comments on the tests should have a blurb which describes at a high-level what is being tested. Ideally it would also state the "why" for the test, though that is sometimes hard to define for happy path. For example:

```
    /// Happy path test for POST /biome/register
    ///
    /// When registering a user with valid inputs, we expect the request to succeed and for the
    /// response to contain the username of the user which was just created.
    ///
    /// Procedure
    ///
    /// 1) Register user via POST /biome/register
    /// 2) Verify that the request was successful
    /// 3) Verify that the username of the user created matches
```

**Pull request merged**